core/mm/core_mmu.c

```diff
@@ -451,30 +501,57 @@ static bool pbuf_is_nsec_ddr(paddr_t pbuf __unused, size_t len __unused)
 #ifdef CFG_SECURE_DATA_PATH
 static bool pbuf_is_sdp_mem(paddr_t pbuf, size_t len)
 {
-	return pbuf_is_special_mem(pbuf, len, phys_sdp_mem_begin,
-				   phys_sdp_mem_end);
+	bool is_sdp_mem = false;
+
+	if (sec_sdp.size)
+		is_sdp_mem = core_is_buffer_inside(pbuf, len,
+						   sec_sdp.paddr,
+						   sec_sdp.size);
+
+	if (!is_sdp_mem)
+		is_sdp_mem = pbuf_is_special_mem(pbuf, len, phys_sdp_mem_begin,
+						 phys_sdp_mem_end);
+
+	return is_sdp_mem;
+}
+
+static struct mobj *core_sdp_mem_alloc_mobj(paddr_t pa, size_t size)
+{
+	struct mobj *mobj = NULL;
+
+	mobj = mobj_phys_alloc(pa, size,
+			       TEE_MATTR_MEM_TYPE_CACHED,
```

Comment:
prefer spread over less lines if possible (not a strong opinion, but more consistent with most optee_os implementation).
```c
	mobj = mobj_phys_alloc(pa, size, TEE_MATTR_MEM_TYPE_CACHED,
				CORE_MEM_SDP_MEM);
```
or 
```c
static struct mobj *core_sdp_mem_alloc_mobj(paddr_t pa, size_t size)
{
	struct mobj *mobj = mobj_phys_alloc(pa, size, TEE_MATTR_MEM_TYPE_CACHED,
					     CORE_MEM_SDP_MEM);
					    
	if (!mobj)
		panic("can't create SDP physical memory object");

	return mobj;			       
}
```